lib/spack/docs/pipelines.rst

```diff
@@ -245,7 +280,21 @@ runners known to the gitlab instance.  For Docker executor type runners, the
 as well as an ``entrypoint`` to override whatever the default for that image is).
 For other types of runners the ``variables`` key will be useful to pass any
 information on to the runner that it needs to do its work (e.g. scheduler
-parameters, etc.).
+parameters, etc.).  Any ``variables`` provided here will be added, verbatim, to
+each job, unless the value takes on a special form.  If the value has the form
+``$env:<var_value>``, then that variable will be inserted with ``var_value``
+first sampled from the environment during job generation.  This can be useful
+for propagating generation-time variables which normally would not be available
+child build jobs.
+
+The ``runner-attributes`` section also allows users to supply custom ``script``,
+``before_script``, and ``after_script`` sections to be applied to every job
+scheduled on that runner.  This allows users to do any custom preparation or
+cleanup tasks that fit their particular workflow, as well as completely
+customize the rebuilding of a spec if they so choose.  We will never generate
```

Comment:
IMO `never` should be changed to `not` or some similar wording to indicate that this statement applies at _this_ time.

Reply:
Makes sense, will do.